streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -256,13 +253,12 @@ public void punctuate(ProcessorNode node, long timestamp) {
             throw new StreamsException(String.format("Exception caught in punctuate. taskId=%s processor=%s", id,  node.name()), ke);
         } finally {
             processorContext.setCurrentNode(null);
-            currNode = null;
         }
     }
 
 
     public ProcessorNode node() {
-        return this.currNode;
+        return this.processorContext.currentNode();
```

Comment:
I think we do not need this function any more since it is only used in `getStateStore` right?